apps/frontend/src/app/PageTeam/CharacterDisplay/Tabs/TabOptimize/Components/AllowChar.tsx

```diff
@@ -291,7 +291,8 @@ export default function AllowChar({
 
   const onMouseUp = useCallback(() => setMouseUpDetected(true), [])
 
-  const total = locList.length
+  // `total` shouldn't rely on `locList.length` because it gets filtered
+  const total = database.chars.keys.length - 1
```

Comment:
Travelers are multiple characters in the character database, e.g., `TravelerAnemo` so you would need to convert it to a list of locations, which is what `locList` represents.

Reply:
I don't understand the problem you are trying to say. But doesn't the `database.chars` stores every character I have in my `/characters` tab. So it's length should be the total count of characters I have.

-----
The problem with `locList` is that with the current implementation it  changes based on the applied filters Which is causing the problem of negative numbers.

-----
When starting everything looks good
<img width="1329" alt="Screenshot 2024-03-03 at 8 50 44 PM" src="https://github.com/frzyc/genshin-optimizer/assets/44922829/90d1ac43-3312-47be-aaff-934cb9499cb0">

-----
Then maybe someone wants to remove all characters. which causes the `excludedLocations.length` to increase and `useTot` to be `0` which is right.
<img width="1282" alt="Screenshot 2024-03-03 at 8 51 21 PM" src="https://github.com/frzyc/genshin-optimizer/assets/44922829/0f6a0ebd-dd0e-4353-9442-cf3efb82dad4">

-----
Until they decide "nah maybe I should add a certain character" so they use filtering at this point `locList` get reduced to the number to the remaining characters while `excludedLocations` remains the same causing `useTot` to be negative.
<img width="1237" alt="Screenshot 2024-03-03 at 8 51 27 PM" src="https://github.com/frzyc/genshin-optimizer/assets/44922829/a7c453ab-3143-4700-bb89-ec41615bc518">
<img width="1293" alt="Screenshot 2024-03-03 at 8 51 41 PM" src="https://github.com/frzyc/genshin-optimizer/assets/44922829/0f034f65-b6a0-40e4-9aa1-d46c16eb2b50">

-----
Another solution may be to change the `onClose` callback to clear the filters so `locList` size would be reset to origin.

Reply:
I understand this might be confusing. my concern was mainly due to the fact that travelers in GO share artifacts, so when you say "take from TravelerAnemo"(characterkey),  you are really saying "take from traveler"(location key). So when you have a tally of "characters" you can take from, the tally is really counting the "number of locations to take artifacts from"

Reply:
Ok. I Got It :/